atomistics/calculators/ase.py

```diff
@@ -29,93 +52,40 @@ def evaluate_with_ase(
             ase_optimizer_kwargs=ase_optimizer_kwargs,
         )
     elif "optimize_positions_and_volume" in tasks:
-        results["structure_with_optimized_positions_and_volume"] = (
-            optimize_positions_and_volume_with_ase(
-                structure=structure,
-                ase_calculator=ase_calculator,
-                ase_optimizer=ase_optimizer,
-                ase_optimizer_kwargs=ase_optimizer_kwargs,
-            )
+        results[
+            "structure_with_optimized_positions_and_volume"
+        ] = optimize_positions_and_volume_with_ase(
+            structure=structure,
+            ase_calculator=ase_calculator,
+            ase_optimizer=ase_optimizer,
+            ase_optimizer_kwargs=ase_optimizer_kwargs,
         )
     elif "calc_energy" in tasks or "calc_forces" in tasks or "calc_stress" in tasks:
-        if "calc_energy" in tasks and "calc_forces" in tasks and "calc_stress" in tasks:
-            (
-                results["energy"],
-                results["forces"],
-                results["stress"],
-            ) = calc_energy_forces_and_stress_with_ase(
-                structure=structure, ase_calculator=ase_calculator
-            )
-        elif "calc_energy" in tasks and "calc_forces" in tasks:
-            results["energy"], results["forces"] = calc_energy_and_forces_with_ase(
-                structure=structure, ase_calculator=ase_calculator
-            )
-        elif "calc_energy" in tasks and "calc_stress" in tasks:
-            results["energy"], results["forces"] = calc_energy_and_stress_with_ase(
-                structure=structure, ase_calculator=ase_calculator
-            )
-        elif "calc_forces" in tasks and "calc_stress" in tasks:
-            results["energy"], results["forces"] = calc_forces_and_stress_with_ase(
-                structure=structure, ase_calculator=ase_calculator
-            )
-        elif "calc_energy" in tasks:
-            results["energy"] = calc_energy_with_ase(
-                structure=structure, ase_calculator=ase_calculator
-            )
-        elif "calc_forces" in tasks:
-            results["forces"] = calc_forces_with_ase(
-                structure=structure, ase_calculator=ase_calculator
-            )
-        elif "calc_stress" in tasks:
-            results["stress"] = calc_stress_with_ase(
-                structure=structure, ase_calculator=ase_calculator
-            )
+        quantities = []
+        if "calc_energy" in tasks:
+            quantities.append("energy")
+        if "calc_forces" in tasks:
+            quantities.append("forces")
+        if "calc_stress" in tasks:
+            quantities.append("stress")
```

Comment:
Here I see the same problem as the one we discussed [here](https://github.com/pyiron/atomistics/pull/118) and [here](https://github.com/pyiron/atomistics/pull/121). To the very least, these lines appear twice inside atomistics, so they have to be somehow refactored, but more fundamentally, I see a clear connection between tags, tasks (i.e. `calc_XYZ`) and interactive getters. Interactive getters and the tags are now fully conceptualised with the data class, and I think the same should happen for the tasks as well.

Reply:
110%, although fair to break that step into a different PR